### PR TITLE
{AKS} add `networkPluginMode` support

### PR DESCRIPTION
**Pull request by tyler-lloyd**
This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
aks-preview

**Review by FumingZhang — COMMENTED** [review]
The PR looks good to me.

For the failures in the CI, it has been fixed in PR [# 4860](https://github.com/Azure/azure-cli-extensions/pull/4860). Please rebase from main.

For another failure in recording test, please add the recording file to `src/aks-preview/azext_aks_preview/tests/latest/recordings/` for your newly added test case `test_aks_create_with_overlay_network_plugin_mode`.

I queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=55610530&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=a26fe913-e2be-5062-b08e-d15f1acc7ea2&l=727) to test this case and got the following error
> E           Code: CidrStringParseError
> E           Message: Cidr string '' is not in a valid CIDR format
> E           Target: agentPoolProfile.vnetSubnetID'

Seems that if vnet subnet is not set, this would cause an error. Is this expected?

Also, would appreciate it if you could add some note about this new option under the `Pending` section in file `src/aks-preview/HISTORY.rst` in case you do not want to release a version of aks-preview.

**Comment by tyler-lloyd** [comment]
Moving this back to draft since the e2e test won't be successful until more changes have been rolled out on the Azure networking side.

**Review by FumingZhang — APPROVED** [review]
LGTM

Better to add a live test case for the new option.

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -16,6 +16,11 @@ Pending
 * Enable v2 decorator pattern.
 * Fix container name inconsistency for private clusters in kollect command.
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:17]: Also move these pending items to the new version section.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 2]
```diff
@@ -198,6 +198,13 @@
           type: string
           short-summary: The Kubernetes network plugin to use.
           long-summary: Specify "azure" for routable pod IPs from VNET, "kubenet" for non-routable pod IPs with an overlay network, or "none" for no networking configured.
+        - name: --network-plugin-mode
+          type: string
+          short-summary: The network plugin mode to use.
+          long-summary: |
+              Specify "overlay" to use an overlay network built on Azure networking.
+              Differs from Kubenet overlay as it does not require route tables.
+              Must be used with --network-plugin=azure.
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_help.py:207]: Could you add a validator for this point?

File: src/aks-preview/HISTORY.rst (hunk 1 above)
Comment (tyler-lloyd) [reply src/aks-preview/HISTORY.rst:18]: Done

File: src/aks-preview/azext_aks_preview/_help.py (hunk 2 above)
Comment (tyler-lloyd) [reply src/aks-preview/azext_aks_preview/_help.py]: Updated the description. Long term, the plugin mode will be able to be used with other network plugins and we should allow aks-rp to handle the validation instead.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**